File: API/Placeful.Api/Models/DTOs/RequestUserFriendshipDto.cs
namespace Placeful.Api.Models.DTOs;

public class RequestUserFriendshipDto
{
    public String FriendshipInitiatorId { get; set; } = String.Empty;
    public String? FriendshipInitiatorName { get; set; } = String.Empty;
    public String FriendshipReceiverId { get; set; } = String.Empty;
    public String? FriendshipReceiverName { get; set; } = String.Empty;
    public bool FriendshipAccepted { get; set; } = false;
}

File: Frontend/placeful/lib/common/domain/dtos/request_user_friendship_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'request_user_friendship_dto.g.dart';

@JsonSerializable()
class RequestUserFriendshipDto {
  final String friendshipInitiatorId;
  final String? friendshipInitiatorName;
  final String friendshipReceiverId;
  final String? friendshipReceiverName;
  final bool friendshipAccepted;

  RequestUserFriendshipDto({
    required this.friendshipInitiatorId,
    required this.friendshipReceiverId,
    required this.friendshipAccepted,
    this.friendshipInitiatorName,
    this.friendshipReceiverName,
  });

  factory RequestUserFriendshipDto.fromJson(Map<String, dynamic> json) =>
      _$RequestUserFriendshipDtoFromJson(json);

  Map<String, dynamic> toJson() => _$RequestUserFriendshipDtoToJson(this);
}
